momepy/coins.py

```diff
@@ -378,118 +382,51 @@ def _list_to_pairs(in_list):
     return [list(pair) for pair in zip(tmp_list[:-1], tmp_list[1:], strict=True)]
 
 
-def _compute_angle(point1, point2):
-    """Calculates the angle between two points in space."""
-    height = abs(point2[1] - point1[1])
-    base = abs(point2[0] - point1[0])
-    angle = round(math.degrees(math.atan(height / base)), 3)
-    return angle
-
-
-def _compute_orientation(line):
-    """Calculates the orientation of a line segment. Point1 is
-    the lower one on the y-axes and vice versa for Point2.
+def _angle_between_two_lines(line1, line2):
+    """
+    Computes interior angle between 2 lines.
+        input:  line<x> ... list of 2 tuples (x,y);
+                line1 and line2 by definition share one unique tuple
+                (overlap in 1 point)
+        returns:    interior angle in degrees 0<alpha<=180
+                    (we assume that line1!=line2, so alpha=0 not possible)
     """
-    point1 = line[1]
-    point2 = line[0]
-
-    # If the latutide of a point is less and the longitude is more, or
-    # If the latitude of a point is more and the longitude is less, then
-    # the point is oriented leftward and wil have negative orientation.
-    if ((point2[0] > point1[0]) and (point2[1] < point1[1])) or (
-        (point2[0] < point1[0]) and (point2[1] > point1[1])
-    ):
-        return -_compute_angle(point1, point2)
-
-    # if the latitudes are same, the line is horizontal
-    elif point2[1] == point1[1]:
-        return 0
-
-    # if the longitudes are same, the line is vertical
-    elif point2[0] == point1[0]:
-        return 90
-
-    return _compute_angle(point1, point2)
-
-
-def _points_set_angle(line1, line2):
-    """Calculate the acute joining angle between two given set of points."""
-    l1orien = _compute_orientation(line1)
-    l2orien = _compute_orientation(line2)
-
-    if ((l1orien > 0) and (l2orien < 0)) or ((l1orien < 0) and (l2orien > 0)):
-        return abs(l1orien) + abs(l2orien)
 
-    elif ((l1orien > 0) and (l2orien > 0)) or ((l1orien < 0) and (l2orien < 0)):
-        theta1 = abs(l1orien) + 180 - abs(l2orien)
-        theta2 = abs(l2orien) + 180 - abs(l1orien)
-        if theta1 < theta2:
-            return theta1
-        else:
-            return theta2
+    # extract points
+    a, b = tuple(line1[0]), tuple(line1[1])
+    c, d = tuple(line2[0]), tuple(line2[1])
 
-    elif (l1orien == 0) or (l2orien == 0):
-        if l1orien < 0:
-            return 180 - abs(l1orien)
-        elif l2orien < 0:
-            return 180 - abs(l2orien)
-        else:
-            return 180 - (
-                abs(_compute_orientation(line1)) + abs(_compute_orientation(line2))
-            )
+    # assertion: we expect exactly 2 of the 4 points to be identical
+    # (lines touch in this point)
+    points = Counter([a, b, c, d])
```

Comment:
one of (many!!) options to figure out which two of the four points in the vectors (a,b) and (c,d) are identical